pydis_site/apps/api/models/bot/tag.py

```diff
@@ -15,7 +15,7 @@ def is_bool_validator(value: Any) -> None:
         raise ValidationError(f"This field must be of type bool, not {type(value)}.")
 
 
-def validate_tag_embed_fields(fields):
+def validate_tag_embed_fields(fields: dict) -> None:
```

Comment:
Do we want to go with a generic `dict` annotation for these types or the more specific `Dict[str, Union[str, bool]]`? It's probably a bit much to go down to this level of specificity for such a large linting PR, but it's something we can consider as a general guideline for regular contributions/PRs.

```suggestion
def validate_tag_embed_fields(fields: Dict[str, Union[str, bool]]) -> None:
```

Reply:
In this case, normally I would say a more detailed type annotation is reasonable. But if the context is considered, this function is meant to validate data so it may accept any data type. It even checks that each field is of type `Mapping` so the only implicit type constraint this function has is that `fields` is iterable.

Reply:
My reasoning went along the lines of:

- We're validating embed fields here, which, according the Discord API reference, should always have the type `Dict[str, Union[str, bool]]`.
- The tag validation function itself assumes that you pass an object of this form as well.

However, I think your point is that the purpose of a validation function is to validate if the object we pass in holds those and more stringent constraints. Honestly, I think we/I may be over-complicating this a bit, which is also why I originally left these comments as comments and not required changes.